crates/bevy_render/src/texture/ktx2.rs

```diff
@@ -0,0 +1,1999 @@
+
+                    let transcoder = LowLevelUastcTranscoder::new();
+                    for (level, level_data) in levels.iter().enumerate() {
+                        let slice_parameters = SliceParametersUastc {
+                            num_blocks_x: ((original_width + block_width_pixels - 1)
+                                / block_width_pixels)
+                                .max(1),
+                            num_blocks_y: ((original_height + block_height_pixels - 1)
+                                / block_height_pixels)
+                                .max(1),
+                            has_alpha: false,
+                            original_width,
+                            original_height,
+                        };
+
+                        transcoder
+                            .transcode_slice(
+                                level_data,
+                                slice_parameters,
+                                DecodeFlags::HIGH_QUALITY,
+                                transcode_block_format,
+                            )
+                            .map(|transcoded_level| transcoded.push(transcoded_level))
+                            .map_err(|error| {
+                                TextureError::SuperDecompressionError(format!(
+                                    "Failed to transcode mip level {} from UASTC to {:?}: {:?}",
+                                    level, transcode_block_format, error
+                                ))
+                            })?;
+
+                        // Next mip dimensions are half the current, minimum 1x1
+                        original_width = (original_width / 2).max(1);
+                        original_height = (original_height / 2).max(1);
+                    }
+                    texture_format
+                }
+                _ => return Err(error),
+            };
+            levels = transcoded;
+            Ok(texture_format)
+        }
+        _ => Err(error),
+    })?;
+    if !supported_compressed_formats.supports(texture_format) {
+        return Err(TextureError::UnsupportedTextureFormat(format!(
+            "Format not supported by this GPU: {:?}",
+            texture_format
+        )));
+    }
+
+    // Assign the data and fill in the rest of the metadata now the possible
+    // error cases have been handled
+    let mut image = Image::default();
+    image.texture_descriptor.format = texture_format;
+    image.data = levels.into_iter().flatten().collect::<Vec<_>>();
+    image.texture_descriptor.size = Extent3d {
+        width,
+        height,
+        depth_or_array_layers: if layer_count > 1 { layer_count } else { depth }.max(1),
+    };
+    image.texture_descriptor.mip_level_count = level_count;
+    image.texture_descriptor.dimension = if depth > 1 {
+        TextureDimension::D3
+    } else if image.is_compressed() || height > 1 {
```

Comment:
Hmm.

Reply:
Metal at least does not support 1D compressed images, so 1D has to be interpreted as a 2D texture.